tensorboard/plugins/mesh/demo_utils.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utils used for the mesh demo."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow as tf
+
+
+def _parse_vertex(vertex_row):
+  """Parses a line in a PLY file which encodes a vertex coordinates.
+
+  Args:
+    vertex_row: string with vertex coordinates and color.
+
+  Returns:
+    `[3,]` array of vertex coordinates and `[3,]` array with RGB color.
+  """
+  vertex = map(float, vertex_row.strip().split(' '))
+  # The row will contain either just coordinates or RGB/RGBA color in addition
+  # to that.
+  assert len(vertex) == 3 or len(vertex) == 6 or len(vertex) == 7
```

Comment:
I think this assertion is not really necessary since this code is meant to be executed on particular .ply file.

Reply:
? By design, it can be executed on any PLY file; it’s controlled by a
command-line argument.

Reply:
If it’s possible to fail the assertion using only the public APIs for
the module (here, `read_ascii_ply`), then it shouldn’t be an assertion.

Reply:
no, it wasn't intended to run on any PLY file, as I mentioned in a comment. I think adding explicit checks to this reader will just complicate code since there are quite few valid variations of PLY file. I'd let it fail with default errors.

Reply:
It doesn’t have to be able to parse arbitrary PLY files. It’s fine if it
raises an exception on files that other PLY files would accept. But it
cannot have an assertion failure. An assertion failure by definition
means that the _code_ is wrong.

All I’m asking is that

```python
  assert len(vertex) == 3 or len(vertex) == 6 or len(vertex) == 7
```

be changed to

```
  if len(vertex) not in (3, 6, 7):
    raise ValueError('Vertices must have 3, 6, or 7 entries; got: %r' % vertex)
```

Reply:
I removed assertion.